### DISCUSSION TITLE
change the cookie expiration

### DISCUSSION THREAD

**Discussion by krissorl** [orestbida/cookieconsent#128] [Q&A]
Hi, 
Can I have  a question? Is there any way to change the cookie exporation? I need to set the diffeent cookie expiration for the necessary and for the analytics approval.
I didn't found any solution.
Thanks.

Kristian
(I'm new on GitLab.)

**Comment by orestbida** [comment]
Hi @krissorl,

unfortnately this is not possible currently, but is something which I plan on adding for usecases like yours!

**Comment by orestbida** [comment]
@krissorl ,

with v2.7 there's now an option called `cookie_necessary_only_expiration` which can be set alongside `cookie_expiration`. Check #100 for more details.

**Reply by krissorl** [reply]
You're great. 
Thanks for the improvements.
I made some improvements for this option. In my solution the _cookie_expiration_ in config can be also a function with one input parameter with the current settings (changed). This function is then called from _setCookie(...) function.
If you want I can give you my solution...

Thanks.
